### ISSUE TITLE
Handle masking of the glomerulus

### ISSUE THREAD

**Issue by thomashopkins32** [thomashopkins32/HuBMAP#4]
We have 3 different types of annotated masks available:
- blood vessels (our target structure)
- glomerulus
- unsure
For the competition, we will be receiving the glomerulus mask on the test set. This means that any prediction we make in the annotated region can be safely ignored as it will not be counted in the scoring part.

I need to see how that annotation is being passed for scoring on the hidden test set. Does Kaggle discard predictions in the region internally?

What should I do with predictions from my model that fall in the glomerulus during training? I should look into this more but here are some ideas:
- Remove any predictions of the model and blood vessel masks in the annotated region (there are some overlaps in the target structures)
- Predict the glomerulus using the model (only needed if this structure is not available in the test set)
- Add a penalty to the loss for predicting blood vessel structures in the glomerulus regions

**Comment by thomashopkins32** [comment]
Here is one issue with partially overlapping blood vessel masks with the glomerulus masks.

![Figure_1](https://github.com/thomashopkins32/HuBMAP/assets/33433321/796d609e-dde2-401e-b241-088eacafaeaf)

When we load the dataset, we have the coordinates of each polygon. So what we can do is remove the blood vessel polygons that have any (or all) overlapping coordinates with the glomerulus.

The issue remains as to how to avoid penalizing the model for predicting blood vessels in the glomerulus region. We can do something like this:
```
max(prediction_mask - glomerulus_mask, 0)
```
I am not sure what this would do to the training objective if we backprop on this. Maybe we shouldn't accumulate gradients on this by using
```python3
with torch.no_grad():
```
but I do not like this option either. The model still has parameters tied to these output pixels and forcing them to 0 does not make sense for training.

**Comment by thomashopkins32** [comment]
A robust solution to this problem is really to predict the glomerulus regions. This way the model can learn to say "This is the glomerulus so I should not predict blood vessels in this region" rather than penalizing it when it could not know any better.

So the ground truth mask can have the following:
```
0 - background
1 - blood vessel
2 - glomerulus
```
And our model will output 3 channels corresponding to these.

When we build the ground truth masks, we should create the glomerulus masks first then overlay the blood vessel masks on top of it. This will make it so no blood vessels are cut off like in the image above.

**Comment by thomashopkins32** [comment]
This will have an impact on how we can handle the unsure masks. I will reference this issue for discussion on this: #5
